src/guest_memory.rs

```diff
@@ -77,6 +77,14 @@ pub enum Error {
     /// Host virtual address not available.
     #[error("Guest memory error: host virtual address not available")]
     HostAddressNotAvailable,
+    /// The length returned by the callback passed to `try_access` is outside the address range.
+    #[error(
+        "The length returned by the callback passed to `try_access` is outside the address range."
+    )]
+    CallbackOutOfRange,
+    /// The address to be read by `try_access` is outside the address range.
+    #[error("The address to be read by `try_access` is outside the address range")]
+    GuestAddressOverflow,
```

Comment:
Following the InvalidGuestAddress variant, these two should also specify which address is invalid. This should make problems easier to debug.

Reply:
These errors occur due to out of bounds addition, so they cannot return the invalid out of bounds value. Should they contain 2 values that added make the out of bounds value? This seems over-complicated.

Reply:
I was more thinking about using the `addr` that was passed to try_access. That should be enough for gettung started with debugging. We can add it later if needed.

Reply:
I think this is a nit that can be in a follow-up, is this okay?